### PR TITLE
[UTOPIA-917]update pia intake view page

### PR DESCRIPTION
**Pull request by binggaofw**
# Description

This PR includes the following proposed change(s):

- add view part for view intake 

Screen Shot 2023-03-13 at 1 28 21 PM

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

File: src/frontend/src/components/public/PIAFormTabs/intake/index.tsx [hunk 1]
```diff
@@ -37,18 +44,90 @@ export const PIAFormIntake = () => {
...
+                  <div className="col col-md-3">{pia.drafterTitle}</div>
+                  <div className="col col-md-3"></div>
+                  <div className="col col-md-3"></div>
+                  <div className="col col-md-3">{pia.leadTitle}</div>
+                </div>
+                <div className="row">
+                  <div className="col col-md-3">{pia.drafterEmail}</div>
+                  <div className="col col-md-3"></div>
+                  <div className="col col-md-3"></div>
+                  <div className="col col-md-3">{pia.leadEmail}</div>
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/public/PIAFormTabs/intake/index.tsx:97]: why scattered data? You can have row inside a column as well - making it nested.

**Comment by kushal-arora-fw** [comment]
Also, the design doesn't match Figma.

File: src/frontend/src/components/public/PIAFormTabs/intake/index.tsx [hunk 2]
```diff
@@ -2,22 +2,29 @@ import MDEditor from '@uiw/react-md-editor';
...
+export const PIAFormIntake = ({ isReadOnly = false }: IntakeFormProps) => {
   const [pia, piaStateChangeHandler] =
     useOutletContext<[IPiaForm, PiaStateChangeHandlerType]>();
 
   const [intakeForm, setIntakeForm] = useState<IPiaFormIntake>(
     exportIntakeFromPia(pia),
   );
 
+  const [piaMinistryFullName, setPiaMinistryFullName] = useState(
+    MinistryList.filter((item) => item.value === pia.ministry)[0].label || '',
```
[the remark this answers is not in the document]
Comment (binggaofw) [reply src/frontend/src/components/public/PIAFormTabs/intake/index.tsx]: the filter function return an array 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

**Comment by binggaofw** [comment]
@kushal-arora-fw  i update the padding for this comp, please take a look, if not ok, tomorrow morning we can pair programming 
Screen Shot 2023-03-13 at 4 51 36 PM

File: src/frontend/src/components/public/PIAIntakeGeneralInformation/index.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
+import { useState } from 'react';
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/public/PIAIntakeGeneralInformation/index.tsx:1]: How about putting this component inside intake directory, and renaming it to viewGeneralInformation.tsx

File: src/frontend/src/components/public/PIAIntakeGeneralInformation/index.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+              <div className="col col-md-3">{pia.mpoName}</div>
+            </div>
+            <div className="row">
+              <div className="col col-md-9">{pia.drafterTitle}</div>
+              <div className="col col-md-3">{pia.leadTitle}</div>
+            </div>
+            <div className="row">
+              <div className="col col-md-9">{pia.drafterEmail}</div>
+              <div className="col col-md-3">{pia.leadEmail}</div>
+            </div>
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/public/PIAIntakeGeneralInformation/index.tsx:46]: Still the same as prev comment. No nested row-col layout leveraged.

File: src/frontend/src/components/public/PIASubHeader/index.tsx [hunk 5]
```diff
@@ -27,6 +28,13 @@ function PIASubHeader({
   const [downloadError, setDownloadError] = useState('');
   const { pathname } = useLocation();
   const nextStepAction = pathname?.split('/').includes('nextSteps');
+  secondaryButtonText = pathname?.split('/').includes('view')
+    ? 'Edit'
+    : ' Save';
+
+  primaryButtonText = pathname?.split('/').includes('view')
+    ? 'Submit'
+    : 'Submit PIA intake';
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/public/PIASubHeader/index.tsx:37]: This logic if the page is a view or edit mode - should reside in the parent - PIAForm page

File: src/frontend/src/pages/PIAForm/index.tsx [hunk 6]
```diff
@@ -268,9 +268,8 @@ const PIAFormPage = () => {
 
   const handleSaveChanges = () => {
     // By default set the status to Incomplete
-    const piaStatus = pia?.status ? pia.status : PiaStatuses.INCOMPLETE;
-    const modalType =
-      piaStatus === PiaStatuses.EDIT_IN_PROGRESS ? 'edit' : 'save';
+    // const piaStatus = pia?.status ? pia.status : PiaStatuses.INCOMPLETE;
+    const modalType = pathname?.split('/').includes('view') ? 'edit' : 'save';
```
Comment (kushal-arora-fw) [inline src/frontend/src/pages/PIAForm/index.tsx:272]: Same logic as above. Should be a state and passed on to the children.

File: src/frontend/src/components/public/PIAIntakeGeneralInformation/index.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+import { IPiaForm } from '../../../types/interfaces/pia-form.interface';
+import { dateToString } from '../../../utils/date';
+
+interface IGeneralInformationProps {
+  pia: IPiaForm;
+}
+
+const PIAIntakeGeneralInformation = ({ pia }: IGeneralInformationProps) => {
+  const [piaMinistryFullName, setPiaMinistryFullName] = useState(
+    MinistryList.filter((item) => item.value === pia.ministry)[0].label || '',
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/public/PIAIntakeGeneralInformation/index.tsx:12]:
```suggestion
    MinistryList.find((item) => item.value === pia.ministry)?.label || '',
```

**Review by Bottle7 — DISMISSED** [review]

File: src/frontend/src/components/public/PIASubHeader/interfaces.ts [hunk 8]
```diff
@@ -1,12 +1,14 @@
 import { MouseEventHandler } from 'react';
-import { ILastSaveAlterInfo } from '../../../pages/PIAIntakeForm';
+import { ILastSaveAlterInfo } from '../../../pages/PIAForm';
 import { IPiaForm } from '../../../types/interfaces/pia-form.interface';
 
 export interface PIASubHeaderProps {
   pia: IPiaForm;
   secondaryButtonText?: string;
   primaryButtonText?: string;
+  mode: string;
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/public/PIASubHeader/interfaces.ts:9]:
```suggestion
  mode: PiaFormOpenMode;
```

**Review by kushal-arora-fw — APPROVED** [review]
minor comment

**Review by Bottle7 — DISMISSED** [review]

**Review by kushal-arora-fw — APPROVED** [review]

**Review by kushal-arora-fw — APPROVED** [review]

**Comment by binggaofw** [comment]
will create a subtask to add dropdown manual update status

**Pull request merged**